android/app/src/main/java/com/dismoi/scout/floating/FloatingService.kt

```diff
@@ -4,139 +4,252 @@ import android.app.Service
 import android.content.Intent
 import android.graphics.PixelFormat
 import android.os.*
+import android.util.Log
 import android.view.Gravity
 import android.view.LayoutInflater
 import android.view.View
 import android.view.WindowManager
+import androidx.annotation.RequiresApi
+import androidx.lifecycle.MutableLiveData
+import com.dismoi.scout.R
 import com.dismoi.scout.floating.layout.*
 import com.dismoi.scout.floating.layout.Message
+import okhttp3.*
+import org.json.JSONObject
+import java.io.IOException
 
+enum class State {
+  HIDDEN, BUBBLE, MESSAGE
+}
 
 class FloatingService : Service() {
+  private val TAG = "FloatingService"
+  private lateinit var handler: Handler
+
+  private lateinit var windowManager: WindowManager
+  private lateinit var inflater: LayoutInflater
   private val binder = FloatingServiceBinder()
-  private var bubblesTrash: Trash? = null
-  private var windowManager: WindowManager? = null
-  private var layoutCoordinator: Coordinator? = null
 
-  override fun onBind(intent: Intent): IBinder? {
+  private var noticesIds: List<Int> = listOf()
+  private var liveNotices: List<MutableLiveData<JSONObject>> = mutableListOf()
+
+  private lateinit var bubbleView: Bubble
+  private lateinit var bubbleLayoutParams: WindowManager.LayoutParams
+
+  private lateinit var messageView: Message
+  private lateinit var messageLayoutParams: WindowManager.LayoutParams
+
+  private lateinit var trashView: Trash
+  private lateinit var trashLayoutParams: WindowManager.LayoutParams
+
+  private var state: State = State.HIDDEN
+
+  @RequiresApi(Build.VERSION_CODES.O)
+  override fun onCreate() {
+    super.onCreate()
+
+    windowManager = getSystemService(WINDOW_SERVICE) as WindowManager
+    inflater = getSystemService(LAYOUT_INFLATER_SERVICE) as LayoutInflater
+    handler = Handler(Looper.getMainLooper())
+
+    createBubbleView()
+    createMessageView()
+    createTrashView()
+
+    Log.d(TAG, "Service created")
+  }
+
+
+  override fun onBind(intent: Intent?): IBinder {
     return binder
   }
 
-  override fun onUnbind(intent: Intent): Boolean {
-    return super.onUnbind(intent)
+  // TODO Proper typed an mapped object !
+  @RequiresApi(Build.VERSION_CODES.N)
+  fun showNotices(noticesIds: List<Int>) {
```

Comment:
Maybe some part of this function's logic could be handled in the `ContextMatcherService` to reduce the responsabilty of this service to *presentation* only.

Reply:
We do not match here, but we do still have fetching that should be moved to repository … Again, will be done after #173